opentelemetry-sdk/tests/trace/export/test_export.py

```diff
@@ -148,6 +148,33 @@ def test_flush(self):
 
         span_processor.shutdown()
 
+    def test_multithreading(self):
+        # Should test that shutdown and force_flush can be called concurrently without
+        # causing racing conditions
+
+        spans_names_list = []
+
+        my_exporter = MySpanExporter(destination=spans_names_list)
+        span_processor_to_flush = export.BatchExportSpanProcessor(my_exporter)
+        span_processor_to_shutdown = export.BatchExportSpanProcessor(
+            my_exporter
+        )
+
+        span_names0 = ["xxx", "bar", "foo"]
+        span_names1 = ["yyy", "baz", "fox"]
+
+        for name in span_names0:
+            _create_start_and_end_span(name, span_processor_to_flush)
+
+        for name in span_names1:
+            _create_start_and_end_span(name, span_processor_to_shutdown)
+
+        # test consecutive calls to force_flush() and shutdown() on separate threads
+        span_processor_to_shutdown.shutdown()
+        self.assertTrue(span_processor_to_flush.force_flush())
```

Comment:
@lzchen is this what you meant by:
> I believe they are referring to the behaviour for `shutdown` and `force_flush` when they are called at the same time in different threads (e.g. multiple BatchSpanProcessors)

It looks like `shutdown()` would block until its done exporting https://github.com/open-telemetry/opentelemetry-python/blob/b085f376e0b0bd1c76e8ff4e3f09e995927f7311/opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py#L253-L259

So they wouldn't be called at the same time unless you actually did it in separate threads. Even if you did it in separate threads, the test might be flaky since you can't guarantee the order of execution.

Reply:
I think @Oberon00 's [comment](https://github.com/open-telemetry/opentelemetry-python/pull/389#discussion_r373401816) was referring to concurrency cases in general (calling `force_flush` manually while the span processor is exporting, calling `shutdown` while the processor is exporting, etc.).

Reply:
^ is this not tested in the way the test has been written currently @lzchen ? As I am calling force_flush and shutdown() manually. unless there is something I am not understanding.

Reply:
They are not called at the same time though, since your call to `shutdown()` doesn't return until it is done exporting